### PR TITLE
Add Hoarder App

### PR DESCRIPTION
**Pull request by mos3abof**
Add Hoarder App

To ensure your issue is dealt with swiftly, please check the following (check the boxes `[x]`):

- [x] Submit one item per pull issue. This eases reviewing and speeds up inclusion.
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [x] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding was first released more than 4 months ago.
- [x] Any software project you are adding has working installation instructions.

Some useful information for the reviewer:

- [First release](https://github.com/hoarder-app/hoarder/releases/tag/v0.1.0) was Feb 20th, 2024.
- [Latest commit](https://github.com/hoarder-app/hoarder/commit/219329b2a60d3aa0b8cda50318ebb4851f12dbdd) is June 9th, 2024.
- [Installation steps](https://docs.hoarder.app/installation/docker)

### CODE REVIEW

File: software/hoarder-app.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - GNU Affero General Public License v3.0
+platforms:
+  - NextJS for the web app. Using app router.
+  - Drizzle for the database and its migrations.
+  - NextAuth for authentication.
+  - tRPC for client->server communication.
+  - Puppeteer for crawling the bookmarks.
+  - OpenAI because AI is so hot right now.
+  - BullMQ for scheduling the background jobs.
+  - Meilisearch for the full content search.
```
Comment (kokomo123) [inline software/hoarder-app.yml:15]:
```suggestion
  - Docker
```
This should all that be needed under platforms, the entire stack is not needed

File: software/hoarder-app.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: "Hoarder App"
+website_url: "https://hoarder.app/"
+source_code_url: "https://github.com/hoarder-app/hoarder"
+description: "A self-hostable bookmark-everything app with a touch of AI for the data hoarders out there."
+licenses:
+  - GNU Affero General Public License v3.0
```
Comment (kokomo123) [inline software/hoarder-app.yml:6]:
```suggestion
  - AGPL-3.0
```
Fixing the license

Comment (mos3abof) [reply software/hoarder-app.yml]: Committed the suggested change.

File: software/hoarder-app.yml (hunk 1 above)
Comment (mos3abof) [reply software/hoarder-app.yml]: Committed the suggested change.

**Review by kokomo123 — APPROVED** [review]
LGTM, pending other thoughts from collaborators. Thanks for your contribution!

**Review by Ki-er — APPROVED** [review]

**Review by nodiscc — APPROVED** [review]

**Comment by mos3abof** [comment]
@nodiscc, @Ki-er, @kokomo123, and @Rabenherz112, thank you all for the reviews. 🙏

Is there anything I need to do to unblock getting this PR merged?

**Pull request merged**